src/client.ts

```diff
@@ -1383,11 +1304,9 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
             } else {
                 this.deviceId = opts.deviceToImport.deviceId;
                 this.credentials.userId = opts.deviceToImport.userId;
-                // will be used during async initialization of the crypto
-                this.exportedOlmDeviceToImport = opts.deviceToImport.olmDevice;
```

Comment:
Should we ditch `opts.deviceToImport` also? Since `exportedOlmDeviceToImport` was only used in the legacy crypto, I suppose it makes no sense to only import the `userId` and the `deviceId`